### ISSUE TITLE
clang-cuda compile issues

### ISSUE THREAD

**Issue by psychocoderHPC** [ComputationalRadiationPhysics/picongpu#3294]
Including some boost header when compiling with clang-cuda produce compile issues:

```
In file included from <built-in>:1:
In file included from /usr/lib/llvm-9/lib/clang/9.0.0/include/__clang_cuda_runtime_wrapper.h:36:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/cmath:45:
In file included from /usr/include/math.h:290:
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:53:13: error: exception specification in declaration does not match previous declaration
__MATHCALL (acos,, (_Mdouble_ __x));
            ^
```

Tested with clang 9 and 10

**Comment by psychocoderHPC** [comment]
workaround, include boost with the following way in each `cpp` file before any other boost includes.

```
#include <boost/version.hpp>
#if (BOOST_VERSION < 106900 && defined(__CUDACC__) &&  defined(__clang__))
#   if defined(__CUDACC__)
#       include <boost/config/compiler/nvcc.hpp>
#   endif
#   if (!defined(__ibmxl__))
#       include <boost/config/compiler/clang.hpp>
#   endif
#   undef __CUDACC__
#   include <boost/config/detail/select_compiler_config.hpp>
#   define __CUDACC__
#endif
```

**Comment by psychocoderHPC** [comment]
This bug is fixed in boost 1.69+ 
https://github.com/boostorg/config/commit/ec3318d47ad6797f250efb26af842b53a012f7ac

**Comment by psychocoderHPC** [comment]
solved with #3295
